### PR TITLE
[RFR] Move a tile and detect victory

### PR DESCRIPTION
**Pull request by Luwangel**
- [x] Move a tile
- [x] Detect the victory

### CODE REVIEW

File: pages/game.js [hunk 1]
```diff
@@ -20,7 +25,26 @@ export default class Index extends Component {
...
+            const newState = {
+                currentGrid: newCurrentGrid,
+                isVictory,
+                turn: turn + 1,
+            };
+
+            this.setState(newState);
+        } catch (error) {
+            console.error(error);
+            return;
```
Comment (jdemangeon) [inline pages/game.js]: This return is useless. It's maybe for eslint?

File: pages/game.js [hunk 2]
```diff
@@ -20,7 +25,26 @@ export default class Index extends Component {
         isVictory: false,
     };
 
-    handleClick = () => {};
+    handleClick = tile => {
```
Comment (jdemangeon) [inline pages/game.js:27]: s/tile/(tile) because it's not an implicit return here (with "()").

File: src/components/tile.js [hunk 3]
```diff
@@ -10,7 +10,9 @@ const Tile = ({ enabled, onClick, tileValue }) => {
 
     return (
         <div
```
Comment (jdemangeon) [inline src/components/tile.js:12]: https://github.com/JedWatson/classnames can let you avoid this kind of ugly condition

File: pages/game.js (hunk 2 above)
Comment (jdemangeon) [reply pages/game.js:27]: They call me julienslint :trollface:

Comment (Kmaschta) [reply pages/game.js:27]: If you want to trigger sedy, you have to add a final / to your sed command @jdemangeon

**Comment by Kmaschta** [comment]
Travis isn't enabled!

File: pages/game.js (hunk 2 above)
Comment (Luwangel) [reply pages/game.js:27]: Unfortunately this is not part of my eslint rules :(

Comment (djhi) [reply pages/game.js:27]: No that's how prettier format it and it's fine

File: pages/game.js [hunk 4]
```diff
@@ -19,7 +24,25 @@ export default class Index extends Component {
...
+
+            const newState = {
+                currentGrid: newCurrentGrid,
+                isVictory,
+                turn: turn + 1,
+            };
+
+            this.setState(newState);
+        } catch (error) {
+            console.error(error);
```
Comment (floo51) [inline pages/game.js:43]: A silenced error is usually a code smell. Either let it throw or add a comment justifying why it's there.

File: src/components/button.js [hunk 5]
```diff
@@ -1,15 +1,23 @@
...
-        <a className={`btn ${color ? color : ''}`}>
-            <i className="material-icons left">{icon}</i>
-            {label}
-        </a>
-    </Link>
-);
+const Button = ({ color, icon, label, path }) => {
+    const buttonClass = ClassNames({
+        btn: true,
+        [`${color}`]: color,
```
Comment (floo51) [inline src/components/button.js]: Isn't``[`${color}`]`` the same as `[color]` ?

**Review by floo51 — CHANGES_REQUESTED** [review]

File: pages/game.js (hunk 4 above)
Comment (Luwangel) [reply pages/game.js:43]: The two methods `findTileByValue` and `move` could return an error. For example, it's the case if the user try to move a wrong tile.

I don't have a fancy way to display this error at the moment, so I prefer to do nothing. Do you think it's a priority ?

Comment (floo51) [reply pages/game.js:43]: Indeed, I don't think it is a priority, you can add a `//TODO` comment explaining why.

**Review by floo51 — APPROVED** [review]

**Pull request merged**